src/Umbraco.Web.UI.Client/src/common/services/clipboard.service.js

```diff
@@ -53,11 +53,40 @@ function clipboardService(notificationsService, eventsService, localStorageServi
         return false;
     }
 
+    function removeKeys(values) {
```

Comment:
We need to make this more generic. This services should not know anything about nested content. The nested content property editor needs to be responsible for clearing these values. To do that we need to allow registering a callback or something where the property editor can say something like "for alias 'nestedcontent' call this method"

I've added a proposal for filling in blank keys here https://github.com/umbraco/Umbraco-CMS/issues/5913#issuecomment-568611712

Reply:
I agree. But we have an architectural problem. 

Cause, if we imagine that Stacked Content also had implemented a copy action, and we had a Stacked Content Element Type has a Nested Content Property, then we need to make sure that the Nested Content entry keys are handled, thought its the Stacked Content Actions handling this.

Might only be a problem with this solution, we might need to look at other solutions on this. The problem is that we cant be sure that any property Angular Code is initialized since the user might never navigate to the content of a copied entry. That's where I thought removing Keys on copy, would be good enough to make the Backend-code know that it should generate keys.

But let me know if you have a better idea.

Reply:
Yes removing the keys and having the back end deal with it is fine and is what I've mentioned in the other task. What I'm saying here is that the clipboard.service.js shouldn't need to know about property editors. And sure it's odd that one property editor might need to know about another one based on these nested things (in the future when we have a streamlined data structure we can prob handle that) but for now, Nested Content should be responsible for clearing it's own key, maybe based on a callback or something but this service shouldn't know what nested content is or any other property editors.

Reply:
Sounds like a good solution, let's get some time dedicated for that solution.

Reply:
Now I have provided a solution for this.